File: estrutura_de_dados/Java/src/metodos_de_busca/CombSort.java
package metodos_de_busca;

public class CombSort {
	public static void comb_sort(int vetor[]) {
		int copiado = 1;
		int intervalo = (vetor.length - 1);
		
		while((intervalo > 1) || (copiado == 1)) {
			intervalo = (intervalo * 10) / 13;
			
			if(intervalo == 9 || intervalo == 10) {
				intervalo = 11;
			}
			
			if(intervalo < 1) {
				intervalo = 1;
			}
			
			copiado  = 0;
			
			for(int i = 0, j = intervalo; j < vetor.length; i++, j++) {
				if(vetor[i] > vetor[j]) {
					int copia = vetor[j];
					vetor[j] = vetor[i];
					vetor[i] = copia;
					copiado = 1;
				}
			}
		}
	}
	
	public static void main(String[] args) {
		int vetor[] = {545, 545, 51, 155, 215, 5161, 511665, 0, 1, 2, 3, 5, 8, 55, 88};
		
		for(int i = 0; i < vetor.length; i++) {
			System.out.print(vetor[i] + " ");
		}
		System.out.println("");
		
		comb_sort(vetor);
		
		for(int i = 0; i < vetor.length; i++) {
			System.out.print(vetor[i] + " ");
		}
	}
}

File: estrutura_de_dados/C++/comb_sort.cpp
#include <iostream>

using namespace std;

void comb_sort(int *vetor, unsigned int tam){
    int intervalo = tam;
    unsigned int trocado = 1;

    while(intervalo > 1 || trocado == 1){
        intervalo = (intervalo * 10) / 13;

        if(intervalo == 9 || intervalo == 10){
            intervalo = 11;
        }

        if(intervalo < 1){
            intervalo = 1;
        }

        trocado = 0;

        for(unsigned int i = 0, j = intervalo; j < tam; i++, j++){
            if(vetor[i] > vetor[j]){
                int copia = vetor[i];
                vetor[i] = vetor[j];
                vetor[j] = copia;
                trocado = 1;
            }
        }
    }
}

int main(){
    int vetor[] = {561, 165, 10, 88, 0, 1, 2, 55, 545, 0, -1};
    int tam = 11;

    for(int i = 0; i < tam; i++){
        cout << vetor[i] << " ";
    }
    cout << " " << endl;

    comb_sort(vetor, tam);

    for(int i = 0; i < tam; i++){
        cout << vetor[i] << " ";
    }
}
